### PR TITLE
Pass score from game to end scene

### LINKED ISSUE #36: Pass Score from one Scene to another
+ Send score from GameScene to EndScene. 
  + Update EndScene's HTML to display final score
+ Then restart score during the transition from EndScene to GameScene.

### PR DESCRIPTION
**Pull request by gonzen2310**
## Overview
Pass accumulated score from GameScene to EndScene when the user dies

### PR Checklist
- [x] Fixes #36
- [x] I have run this code to verify it works
~~This PR includes tests for the code change~~

### Notes
Screen Shot 2019-06-27 at 7 05 53 PM

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "@codecademy/40phaser",
```
Comment (JoshuaKGoldberg) [inline package-lock.json]: Could you please `git checkout master -- package-lock.json`? We should resolve this separately. 

The change in name from a previous value to what it is now in `package.json` makes me think the changes here are correct, just out of place.

File: src/scenes/game.js [hunk 2]
```diff
@@ -19,7 +19,8 @@ export default class GameScene extends Phaser.Scene {
         });
     }
 
-    create() {
+    create(data) {
+        score = data.score ? data.score : 0;
```
Comment (JoshuaKGoldberg) [inline src/scenes/game.js]: 😉 

```suggestion
        score = data.score || 0;
```

Comment (JoshuaKGoldberg) [inline src/scenes/game.js]: Also, now that you're always setting `score` to default to `0` here, you can remove the `= 0;` from above. It's unnecessary now _(I think? right?)_

File: src/scenes/end.js [hunk 3]
```diff
@@ -7,19 +7,17 @@ export default class EndScene extends Phaser.Scene {
...
 
-    create() {
+    create(data) {
         elements.appEnd.style.visibility = "visible";
         elements.appEnd.style.display = "flex";
         elements.appEnd.style.marginTop = "10%";
-
-        // TODO: Update display score once gameState is implemented
-        // elements.endScoreDisplay.innerHTML = `Score: ${}`
+        elements.endScoreDisplay.innerHTML = `Score: ${Math.trunc(data.score)}`;
```
Comment (JoshuaKGoldberg) [inline src/scenes/end.js]: Why the `Math.trunc`? I feel like you could remove it, right?
```suggestion
        elements.endScoreDisplay.innerHTML = `Score: ${data.score}`;
```

Comment (gonzen2310) [reply src/scenes/end.js]: This happens when I remove Math.trunc(data.score).

Screen Shot 2019-06-28 at 10 01 05 AM

Should keep it?

File: src/scenes/game.js (hunk 2 above)
Comment (gonzen2310) [reply src/scenes/game.js]: Agree. I'll remove

File: src/scenes/end.js (hunk 3 above)
Comment (JoshuaKGoldberg) [reply src/scenes/end.js]: LOL. Why is that happening? Can you find where the score is becoming this odd thing?

Comment (LinKCoding) [reply src/scenes/end.js]: That's interesting, I thought there that `score` was being `Math.floor()`ed somewhere in **game.js**

File: src/scenes/game.js (hunk 2 above)
Comment (LinKCoding) [reply src/scenes/game.js]: I'm wondering when `score` is not 0 at the start of a `GameScene`

File: src/scenes/end.js (hunk 3 above)
Comment (gonzen2310) [reply src/scenes/end.js]: I'll try to find out

**Pull request merged**